### DISCUSSION TITLE
Chinese display is garbled in OSG3.4 and osgEarth2.7

### DISCUSSION THREAD

**Discussion by XianranZ** [pelicanmapping/osgearth#2396] [Support Q&A] [answered]
When osgearth is adding a vector file, what encoding does it use to read the attribute table data of this vector file?
I have a vector file that stores cities' names in its attribute table, but when I add them to the earth's surface with osgearth, the label display is all garbled.
Here is my code
```
osgEarth::Drivers::OGRFeatureOptions featureOption;
featureOption.url() = qstrFilePath.toStdString();
osgEarth::Style style;
osgEarth::Symbology::PointSymbol* ps = style.getorcreateSymbol<osgEarth::Symboloay::PointSymbol>();
ps->size() = 3.0f;
ps->fill() = osgEarth::Color::Red;
osgtarth::Drivers::FeatureGeomModelOptions geomOptions;
geomOptions.feature0ptions() = feature0ption;
geomOptions.styles() = new osgEarth::Symbology::StyleSheet();
geom0ptions.styles()->addStyle( style );
geomOptions.enablelighting() = false;
osgEarth: :ModellayerOptions layerOptions("my features", geomOptions);
m_mapNode->getMap()->addModelLayer (new oseEarth::ModelLayer(layerOptions));
osgEarth::Style labelStyle;
osgEarthi:Synbology::Textsymbol* text = labelStyle.getOrCreateSymbol<osgEarth::Symbology: :TextSymbol>();
text->content() = osgEarth::StringExpression("Name");
text->removeDuplicateLabels() = false;
text->size()= 16.0f;
text->alignment() = osgEarth::Symbology::TextSymbol::ALIGN_LEFT_CENTER;
text->fill()->color() = osgEarth: : Symbology::Color::White;
text->halo()->color() = osgEarth: : Symbology::Color::DarkGray;
text->font = "simhei.ttf";
text->encoding() = osgEarth: : Symbology: :TextSymbol::ENCOOING_UTF8;
text->pixelOffset() = osg::Vec2s(10, 10);
osgEarth::Drivers::FeatureGeomModelOptions geomOptions1;
geomOptions1.featureOptions = featureOption;
geomOptions1.styles() = new osgEarth::Symbology::StyleSheet();
geomOptions1.styles()->addStyle( labelStyle);
m_mapNode->getMap()->addModelLayer( new osgEarth::ModelLayer("labels", geomOptions1));
```
Here is the display
![IMG_20231207_165731.jpg](https://github.com/gwaldron/osgearth/assets/46214062/b8d87c5f-17ae-4849-a26a-34a6fb81622a)
I guess osgearth calls gdal somewhere and gets the attribute information of the vector file, and I want to make the program get the correct attribute data by specifying the encoding method when gdal reads the vector file.But I didn't find the place in osgearth source code where it reads the vector file.Does anyone know about it?
Or, is there a better solution?
Oh yeah, my vector file attribute table is encoded in GBK
Thanks a lot!

**Comment by gwaldron** [comment]
I could be completely wrong, but I thought UTF8 encoded feature attributes were enabled by this line in Registry.cpp:
```c++
CPLSetConfigOption("GDAL_FILENAME_IS_UTF8","YES");
```
That is protected by an `#ifdef OSG_USE_UTF8_FILENAME` that you may want to ensure is set.

**Reply by XianranZ** [reply, marked as answer]
Thanks very much! Your answer is very helpful for me.
But it seems that it works before adding the below sentence.

`CPLSetConfigOption("SHAPE_ENCODING","CP936");`

After adding this, recompile the osgEarth and it works.
